MoreLinq/SkipLastWhile.cs

```diff
@@ -0,0 +1,65 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2024 Andy Romero (armorynode). All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Removes elements from the end of a sequence as long as a specified condition is true.
+        /// </summary>
+        /// <typeparam name="T">Type of the source sequence.</typeparam>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="predicate">The predicate to use to remove items from the tail of the sequence.</param>
+        /// <returns>
+        /// An <see cref="IEnumerable{T}"/> containing the source sequence elements except for the bypassed ones at the end.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">The source sequence is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentNullException">The predicate is <see langword="null"/>.</exception>
+
+        public static IEnumerable<T> SkipLastWhile<T>(this IEnumerable<T> source, Func<T, bool> predicate)
+        {
+            if (source == null) throw new ArgumentNullException(nameof(source));
+            if (predicate == null) throw new ArgumentNullException(nameof(predicate));
+
+            return _(source, predicate);
+
+            static IEnumerable<T> _(IEnumerable<T> source, Func<T, bool> predicate)
+            {
+                var queue = new Queue<T>();
+                foreach (var item in source)
+                {
+                    if (predicate(item))
+                    {
+                        queue.Enqueue(item);
+                    }
+                    else
+                    {
+                        while (queue.Count > 0)
+                        {
+                            yield return queue.Dequeue();
+                        }
+                        yield return item;
+                    }
+                }
+            }
```

Comment:
I do like the idea of optimizing for specific types instead of treating them all the same, but I do have a question about `.TryAsListLike()`. Would checking for the count on line `44` potentially cause a multiple enumeration if the count is greater than 0 and `IterateList` is called? It's probably a non-issue, but I would like to know what your thoughts are on the potential performance implications🙂

Reply:
> Would checking for the count on line `44` potentially cause a multiple enumeration if the count is greater than 0 and `IterateList` is called?

No. The `Count` is checked without any iteration _if_ the runtime type supports querying the count:

https://github.com/morelinq/MoreLINQ/blob/03a160ceaa0942f46cf585894a3c6e53df040076/MoreLinq/ListLike.cs#L61-L68

`TestingSequence` also has a built-in check against multiple iterations:

https://github.com/morelinq/MoreLINQ/blob/03a160ceaa0942f46cf585894a3c6e53df040076/MoreLinq.Test/TestingSequence.cs#L92

`maxEnumerations` is set to a default of 1 in `TestingSequence.Of`:

https://github.com/morelinq/MoreLINQ/blob/03a160ceaa0942f46cf585894a3c6e53df040076/MoreLinq.Test/TestingSequence.cs#L29-L33

Hope this clears up the mystery.